app/actions/remote/channel.ts

```diff
@@ -1055,3 +1062,43 @@ export async function searchAllChannels(serverUrl: string, term: string, archive
         return {error};
     }
 }
+
+export const updateChannelNotifyProps = async (serverUrl: string, channelId: string, props: Partial<ChannelNotifyProps>) => {
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    try {
+        const userId = await getCurrentUserId(database);
+        const notifyProps = {...props, channel_id: channelId, user_id: userId};
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore only feed the changed notification property
+        await client.updateChannelNotifyProps(notifyProps);
+
+        const channelSettings = await queryMyChannelSettingsByIds(database, [channelId]).fetch();
+        const currentSettings = channelSettings?.[0];
+
+        if (currentSettings && !compareNotifyProps(notifyProps, currentSettings.notifyProps)) {
+            await database.write(async () => {
+                await currentSettings.update((c) => {
+                    c.notifyProps = {...props, ...currentSettings.notifyProps};
```

Comment:
Since currentSettings.nofifyProps is in second place, doesn't this overwrite any change that may have appear on props? That doesn't look intended.

Also, why query the current settings? I guess it is mainly to do the comparison, but shouldn't we use here `c.notifyProps` instead of `currentSettings.notifyProps`?

Reply:
This has been updated